nifi-nar-bundles/nifi-zendesk-bundle/nifi-zendesk-common/src/main/java/org/apache/nifi/common/zendesk/util/ZendeskRecordPathUtils.java

```diff
@@ -0,0 +1,169 @@
+        final RecordPath recordPath = RecordPath.compile(pathValue);
+        final RecordPathResult result = recordPath.evaluate(record);
+        final List<FieldValue> fieldValues = result.getSelectedFields().collect(toList());
+        final FieldValue fieldValue = getMatchingFieldValue(recordPath, fieldValues);
+
+        if (fieldValue.getValue() == null || fieldValue.getValue() == NULL_VALUE) {
+            return null;
+        }
+
+        return getFieldValue(recordPath, fieldValue);
+    }
+
+    /**
+     * The method checks the field's type and filters out every non-compatible type.
+     * @param recordPath path to the requested field
+     * @param fieldValue record field
+     * @return value of the record field
+     */
+    private static String getFieldValue(final RecordPath recordPath, FieldValue fieldValue) {
+        final RecordFieldType fieldType = fieldValue.getField().getDataType().getFieldType();
+
+        if (fieldType == RecordFieldType.RECORD || fieldType == RecordFieldType.ARRAY || fieldType == RecordFieldType.MAP) {
+            throw new ProcessException(String.format("The provided RecordPath [%s] points to a [%s] type value", recordPath, fieldType));
+        }
+
+        if (fieldType == RecordFieldType.CHOICE) {
+            final ChoiceDataType choiceDataType = (ChoiceDataType) fieldValue.getField().getDataType();
+            final List<DataType> possibleTypes = choiceDataType.getPossibleSubTypes();
+            if (possibleTypes.stream().anyMatch(type -> type.getFieldType() == RecordFieldType.RECORD)) {
+                throw new ProcessException(String.format("The provided RecordPath [%s] points to a [CHOICE] type value with Record subtype", recordPath));
+            }
+        }
+
+        return String.valueOf(fieldValue.getValue());
+    }
+
+    /**
+     * The method checks if only one result were received for the give record path.
+     * @param recordPath path to the requested field
+     * @param resultList result list
+     * @return matching field
+     */
+    private static FieldValue getMatchingFieldValue(final RecordPath recordPath, final List<FieldValue> resultList) {
+        if (resultList.isEmpty()) {
+            throw new ProcessException(String.format("Evaluated RecordPath [%s] against Record but got no results", recordPath));
+        }
+
+        if (resultList.size() > 1) {
+            throw new ProcessException(String.format("Evaluated RecordPath [%s] against Record and received multiple distinct results [%s]", recordPath, resultList));
+        }
+
+        return resultList.get(0);
+    }
+
+    /**
+     * Adds a new node on the provided path with the give value to the request object.
+     * @param baseNode base object where the new node will be added
+     * @param path path of the new node
+     * @param value value of the new node
+     */
+    public static void addNewNodeAtPath(ObjectNode baseNode, JsonPointer path, JsonNode value) {
```

Comment:
This method could be simplified by extracting several part into new submethods.
I recommend extracting parent node creation and setting the node into new methods.
This would also eliminate direct recursion.